File: ex_lista/src_python/Bissecao.py
'''
Ana Cristina Silva de Oliveira - 11965630
BCC - A ex07 Metodo bissecao
'''
import math

def main():
    a, b = 0,0
    try:
        print("Digite dois valores para o intervalor [a,b]")
        a = int(input())
        b = int(input())
    except:
        print("Voce nao digitou um valor numerico. Tente novamente")
        exit()

    c = 0.0
    iteracoes = 0
    
    while (iteracoes >= 0):
        c = (a+b)/2
        f_c = math.pow(c,3) - math.pow(c,2) - 13*c + 8;
        f_a = math.pow(a,3) - math.pow(a,2) - 13*a + 8;
        
        if f_c == 0 or b-a < 0.0000001:
            print("A resposta é {}".format(c))
            break
        
        if f_a*f_c < 0:
            b = c
        else:
            a = c
            
        iteracoes+=1
    
    print("iteracoes: {}".format(iteracoes))
    
    
if __name__ == "__main__":
    main()

File: ex_lista/src_java/Bissecao.java
/* Ana Cristina Silva de Oliveira - NUSP 11965630
   Turma: BCC - A 
   Exercício 07 Método da bisseção
*/

import java.io.IOException;

public class Bissecao {
    public static void main(String[] args) throws IOException{
        double a = 0, b = 0;

        System.out.println("Digite dois valores para o intervalo [a,b]: ");
        a = EntradaTeclado.leDouble();
        b = EntradaTeclado.leDouble();

        double c = 0.0;
        
        int i = 0; // contador de iterações
        
        while (i >= 0) {
            c = (a+b)/2; // valor medio do intervalo [a,b]
            double f_c = Math.pow(c,3) - Math.pow(c,2) - 13*c + 8;
            double f_a = Math.pow(a,3) - Math.pow(a,2) - 13*a + 8;
            
            if (f_c == 0 || b - a < 0.0000001) {
                System.out.printf("A resposta é c = %f\n", c);
                break;
            }

            if (f_c * f_a < 0) {
                b = c;
            } else { 
                a = c;
            }

            i++;

        }
        
        System.out.printf("iteracoes: %d", i);
    
    }
}
